src/libData/AccountData/AccountStore.cpp

```diff
@@ -817,8 +817,11 @@ bool AccountStore::MigrateContractStates(
 
     account.SetStorageRoot(dev::h256());
     // invoke scilla checker
-    m_scillaIPCServer->setContractAddressVerRoot(address, scilla_version,
-                                                 account.GetStorageRoot());
+    // prepare IPC with current blockchain info provider.
+    ScillaBCInfoProvider sbcip(getCurBlockNum(), address,
+                               account.GetStorageRoot(), scilla_version);
+    m_scillaIPCServer->setBCInfoProvider(&sbcip);
```

Comment:
I'm not sure it's completely safe to assign the address of a local variable (`sbcip`) for later use in other functions unless it's guaranteed the local variable always still remains in its current scope. Maybe smart pointer or even pass-by-value would be better, even if less efficient.